profiles/views/profile_detail_views.py

```diff
@@ -0,0 +1,74 @@
+from rest_framework.generics import (
+    RetrieveUpdateDestroyAPIView,
+)
+from utils.completeness_counter import completeness_count
+from ..models import SavedCompany, Profile
+from ..permissions import UserIsProfileOwnerOrReadOnly
+from ..serializers.profile_detail_serializers import (
+    ProfileOwnerDetailViewSerializer,
+    ProfileSensitiveDataROSerializer,
+    ProfileDeleteSerializer,
+    ProfileOwnerDetailEditSerializer,
+    ProfileDetailSerializer,
+)
+
+
+class ProfileDetail(RetrieveUpdateDestroyAPIView):
+    """
+    Retrieve, update or delete a profile instance.
+    Retrieve:
+        If user is a person in the profile, full info returned.
+        Else profile info without sensitive data returned.
+        If user is authenticated, he can get sensitive data via query param 'with_contacts'.
+    """
+
+    queryset = (
+        Profile.objects.active_only()
+        .select_related("person")
+        .prefetch_related("regions", "categories", "activities")
+    )
+    permission_classes = [UserIsProfileOwnerOrReadOnly]
+
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        if self.request.user.is_authenticated:
+            saved_companies_pk = frozenset(
+                SavedCompany.objects.filter(
+                    user=self.request.user
+                ).values_list("company_id", flat=True)
+            )
+            context.update({"saved_companies_pk": saved_companies_pk})
+        return context
+
+    def get_serializer_class(self):
+        get_contacts = self.request.query_params.get("with_contacts")
+
+        profile_pk = self.kwargs.get("pk")
```

Comment:
Don't mess with `self.kwargs` when it's not needed.

`self.get_object()` returns the object you need.